packages/shared/components/popups/LegalUpdate.svelte

```diff
@@ -0,0 +1,96 @@
+<script lang="typescript">
+    import { Button, Checkbox, Text, Link } from 'shared/components'
+    import { lastAcceptedTos, lastAcceptedPrivacyPolicy } from 'shared/lib/appSettings'
+    import { localize } from 'shared/lib/i18n'
+    import { closePopup } from 'shared/lib/popup'
+    import { needsToAcceptLatestTos, needsToAcceptLatestPrivacyPolicy, TOS_VERSION, PRIVACY_POLICY_VERSION } from 'shared/lib/app'
+    import { Platform } from 'shared/lib/platform'
+
+    const TOS_LINK = 'https://firefly.iota.org/terms'
+    const PRIVACY_POLICY_LINK = 'https://firefly.iota.org/privacy'
+
+    let checked = false
+    const tos = needsToAcceptLatestTos()
+    const privacyPolicy = needsToAcceptLatestPrivacyPolicy()
+
+    const handleViewTosClick = () => {
+        Platform.openUrl(TOS_LINK)
+    }
+
+    const handleViewPrivPolicyClick = () => {
+        Platform.openUrl(PRIVACY_POLICY_LINK)
+    }
+
+    const handleConfirmClick = () => {
+        if (tos) {
+            lastAcceptedTos.set(TOS_VERSION)
+        }
+        if (privacyPolicy) {
+            lastAcceptedPrivacyPolicy.set(PRIVACY_POLICY_VERSION)
+        }
+
+        closePopup(true)
+    }
+
+    const getTitleText = () => {
+        if (tos && privacyPolicy) {
+            return 'views.legal.title'
+        } else if (tos) {
+            return 'popups.legalUpdate.tosTitle'
+        } else if (privacyPolicy) {
+            return 'popups.legalUpdate.privPolicyTitle'
+        }
+    }
+
+    const getBodyText = () => {
+        if (tos && privacyPolicy) {
+            return 'popups.legalUpdate.tosAndPrivPolicyBody'
+        } else if (tos) {
+            return 'popups.legalUpdate.tosBody'
+        } else if (privacyPolicy) {
+            return 'popups.legalUpdate.privPolicyBody'
+        }
+    }
+
+    const getCheckboxText = () => {
+        if (tos && privacyPolicy) {
+            return 'popups.legalUpdate.tosAndPrivPolicyCheckbox'
+        } else if (tos) {
+            return 'popups.legalUpdate.tosCheckbox'
+        } else if (privacyPolicy) {
+            return 'popups.legalUpdate.privPolicyCheckbox'
+        }
+    }
+
+</script>
+
+<style type="text/scss">
+    ul {
+        display: block;
+        list-style-type: disc;
+        margin-block-start: 1em;
+        margin-block-end: 1em;
+        margin-inline-start: 0px;
+        margin-inline-end: 0px;
+        padding-inline-start: 20px;
+    }
+</style>
```

Comment:
We should move style below the component tags, to stick to our new convention of script, component, style.

Reply:
Done for you :)

Reply:
Ah whoops, I remembered wrong and thought it was script, style, component 😂  thanks!